packages/smooth_app/lib/pages/user_management/sign_up_page.dart

```diff
@@ -0,0 +1,311 @@
+                return null;
+              },
+            ),
+            const SizedBox(height: space),
+            SmoothTextFormField(
+              type: TextFieldTypes.PLAIN_TEXT,
+              controller: _userController,
+              hintText: 'Username',
+              prefixIcon: const Icon(Icons.person),
+              autofillHints: const <String>[AutofillHints.username],
+              validator: (String? value) {
+                if (value == null || value.isEmpty) {
+                  return 'Please enter a username';
+                }
+                return null;
+              },
+            ),
+            const SizedBox(height: space),
+            SmoothTextFormField(
+              type: TextFieldTypes.PASSWORD,
+              controller: _password1Controller,
+              hintText: appLocalizations.password,
+              prefixIcon: const Icon(Icons.vpn_key),
+              autofillHints: const <String>[AutofillHints.password],
+              validator: (String? value) {
+                if (value == null || value.isEmpty) {
+                  return 'Please enter a password';
+                }
+                if (!UserManagementHelper.isPasswordValid(value)) {
+                  return 'Please enter a valid password (at least 6 characters)';
+                }
+                return null;
+              },
+            ),
+            const SizedBox(height: space),
+            SmoothTextFormField(
+              type: TextFieldTypes.PASSWORD,
+              controller: _password2Controller,
+              hintText: 'Confirm Password',
+              prefixIcon: const Icon(Icons.vpn_key),
+              autofillHints: const <String>[
+                AutofillHints.password,
+              ],
+              validator: (String? value) {
+                if (value == null || value.isEmpty) {
+                  return 'Please confirm the password';
+                }
+                if (value != _password1Controller.text) {
+                  return 'The passwords should be identical';
+                }
+                return null;
+              },
+            ),
+            const SizedBox(height: space),
+            Text('Username cannot contains spaces, caps or special characters'),
+            const SizedBox(height: space),
+            // careful with CheckboxListTile and hyperlinks
+            // cf. https://github.com/flutter/flutter/issues/31437
+            ListTile(
+              leading: Checkbox(
+                value: _agree,
+                onChanged: (final bool? value) {
+                  if (value != null) {
+                    setState(() => _agree = value);
+                  }
+                },
+              ),
+              title: RichText(
+                text: TextSpan(
+                  children: <InlineSpan>[
+                    TextSpan(
+                      text: 'I agree to the Open Food Facts ',
+                    ),
+                    TextSpan(
```

Comment:
It's going to work for French and English, but it might not for all languages

Reply:
Anyway that part was a pain in the neck: the display needs to be changed.

I had an `Exception` with "semantics": the checkbox is more or less a large on/off button with one label. If you put inside the checkbox label something that can be clicked, it brings confusion to the "Accessibility" tools. Which text should be said out loud: the whole label? only the non-clickable part?